src/transport/message.rs

```diff
@@ -0,0 +1,114 @@
+use std::collections::HashMap;
+use bytes::{Bytes, BytesMut};
+use protocol::*;
+use types::*;
+use codec::Encode;
+
+#[derive(Debug, Clone)]
+pub struct Message {
+    pub header: Option<Header>,
+    pub delivery_annotations: Option<Annotations>,
+    pub message_annotations: Option<Annotations>,
+    pub properties: Option<Properties>,
+    pub application_properties: Option<HashMap<ByteStr, Variant>>,
+    pub application_data: MessageBody,
+    pub footer: Option<Annotations>
+}
+
+#[derive(Debug, Clone)]
+pub enum MessageBody {
+    Data(Bytes),
+    DataVec(Vec<Bytes>),
+    SequenceVec(Vec<List>),
+    Value(Variant)
+}
+
+const SECTION_PREFIX_LENGTH: usize = 3;
+
+impl Message {
+    pub(crate) fn serialize(self) -> Bytes {
+        let mut dst = BytesMut::with_capacity(self.encoded_size());
+        if let Some(h) = self.header {
+            Section::Header(h).encode(&mut dst);
+        }
+        if let Some(da) = self.delivery_annotations {
+            Section::DeliveryAnnotations(da).encode(&mut dst);
+        }
+        if let Some(ma) = self.message_annotations {
+            Section::MessageAnnotations(ma).encode(&mut dst);
+        }
+        if let Some(p) = self.properties {
+            Section::Properties(p).encode(&mut dst);
+       }
```

Comment:
nit: whitespace

Reply:
mmm?

Reply:
The brace doesn't line up with the `if` two lines above.